lib/SetAPI/generic/SetInterfaceV1.py

```diff
@@ -89,12 +131,82 @@ def _check_save_set_params(self: "SetInterfaceV1", params: dict[str, Any]) -> No
             if not params.get(param):
                 raise ValueError(param_required(param))
 
+        # this should not happen with the current API, but best to check
+        if not set_item_name or set_item_name not in SET_ITEM_NAME_TO_SET_TYPE:
+            err_msg = f"invalid set item name: {set_item_name}"
+            raise ValueError(err_msg)
+
+        # bail out early for RNASeqSampleSets
+        if set_item_name == RNASEQ_SAMPLE:
+            return set_item_name
+
+        # convert the set item name into the set type
+        save_search_set = params.get(SAVE_SEARCH_SET, False)
+        if save_search_set and set_item_name == "genome":
+            set_item_name = GENOME_SEARCH
+
+        list_item_type = "elements" if save_search_set else "items"
+        if list_item_type not in params["data"]:
+            raise ValueError(list_required(list_item_type))
+
+        if "description" not in params["data"]:
+            params["data"]["description"] = ""
+
+        # ensure that empty sets are only present in those set types
+        # that allow them
+        if set_item_name not in ALLOWS_EMPTY_SETS and not params["data"].get(
+            list_item_type
+        ):
+            raise ValueError(no_items(SET_ITEM_NAME_TO_SET_TYPE[set_item_name]))
+
+        if list_item_type == "elements":
+            return set_item_name
```

Comment:
the following checks are related to the `items` array, so obvs they should not be applied if the data structure does not have that array!